test/integ/test_key_pair_authentication.py

```diff
@@ -64,6 +64,49 @@ def test_get_token_from_private_key(input_account, expected_account):
     )
 
 
+@pytest.mark.skipolddriver
+def test_regionless_url_JWT_token_validity(request, db_parameters):
+
+    test_user = "admin"
+
+    db_config = {
+        "account": "amoghorgurl-keypairauth_test_alias.testdns",
+        "user": "admin",
+        "role": "ACCOUNTADMIN",
+        "timezone": "UTC",
+    }
+
+    db_config_with_pw = {
+        "account": "amoghorgurl-keypairauth_test_alias.testdns",
+        "user": "admin",
+        "password": "Password1",
+        "role": "ACCOUNTADMIN",
+        "timezone": "UTC",
+    }
+
+    cnx = snowflake.connector.connect(**db_config_with_pw)
+    cursor = cnx.cursor()
+    cursor.execute(
+        """
+    use role accountadmin
+    """
+    )
+
+    private_key_der, public_key_der_encoded = generate_key_pair(2048)
+
+    cnx.cursor().execute(
+        """
+    alter user {user} set rsa_public_key='{public_key}'
+    """.format(
+            user=test_user, public_key=public_key_der_encoded
+        )
+    )
```

Comment:
There's no reason for your short queries to be this long, also [f-strings](https://www.python.org/dev/peps/pep-0498/)
```suggestion
    cnx.cursor().execute(f"alter user {test_user} set rsa_public_key='{public_key_der_encoded}'")
    )
```
Also indent it if you switch to using a with statement for `cnx`